torch/_inductor/ir.py

```diff
@@ -258,6 +258,19 @@ def get_stride_order(seq: Sequence[Union[int, torch.SymInt, Expr]]) -> Sequence[
     return out
 
 
+def get_new_stride_with_stride_order(
+    stride: Sequence[Union[int, torch.SymInt, Expr]],
+    stride_order: Sequence[int],
+) -> Sequence[Union[int, torch.SymInt, Expr]]:
+    """
+    Get new strides following stride_order
+    """
+    new_stride = [stride[0] for _ in range(len(stride))]
+    for i, elem in enumerate(stride_order):
+        new_stride[stride_order[i]] = stride[elem]
+    return new_stride
```

Comment:
You can't compute a new stride from a stride order without knowing the shapes, so this is giving wrong strides. Probably why all the timm models that use this are having huge performance regressions.

Also, the code here already exists: https://github.com/pytorch/pytorch/blob/81a15cc555942f4a47af81cdc11b9db03a91a3f4/torch/_inductor/ir.py#L3185-L3186

Reply:
fixed in https://github.com/pytorch/pytorch/pull/130956/commits/fe6708eb9ad617309713c33d5a56faa63e41e5d5